### DISCUSSION TITLE
Could not poll the database

### DISCUSSION THREAD

**Discussion by vader-pepe** [firefly-iii/firefly-iii#8564] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [ ] ... there are no open or closed issues that are related to my problem
- [ ] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I followed the instructions on installing firefly-iii with docker (im using podman actually), then I got error could not poll the database when it's started.

### Debug information

Could not poll the database: SQLSTATE[HY000] [2054] Server sent charset (0) unknown to the client. Please, report to the developers (Connection: mysql, SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1)

This error occured in file /var/www/html/app/Support/FireflyConfig.php on line 65 with code 0.

### Expected behaviour

I can see the login page as shown by the demo

### Steps to reproduce

1. clone the repo
2. create docker-compose.yml
3. change the port to 3000 (80 doesn't work for me. does't have to be 3000. can be anything)
4. add 'docker.io/' prefix for image if you using podman
5. run podman-compose up -f docker-compose.yml -d 
6. open localhost:3000

**Comment by vader-pepe** [comment]
possible same issue: [#1366](https://github.com/firefly-iii/firefly-iii/issues/1366)

**Comment by JC5** [comment]
I'm pretty sure the database container failed to start for some reason, check out the logs.

**Reply by vader-pepe** [reply]
![image](https://github.com/firefly-iii/firefly-iii/assets/93415755/efbde355-e03f-41ea-8c29-db41679a9222)

**Reply by vader-pepe** [reply]
![image](https://github.com/firefly-iii/firefly-iii/assets/93415755/0a164320-96b1-43bc-91ff-3922403cb413)
i can connect to the container without issue

**Reply by JC5** [reply]
Sorry, then I don't know what it could be.

**Reply by vader-pepe** [reply]
Could it be WSL problem?

**Reply by Still34** [reply]
I just got hit with the same problem about an hour ago. It was working perfectly fine before then.

**Reply by Still34** [reply, marked as answer]
Specifying an older build of mariadb (specifically 11.2) fixed the problem - seems like something's going on with the latest MariaDB Docker release.

**Comment by vader-pepe** [comment]
> Specifying an older build of mariadb (specifically 11.2) fixed the problem - seems like something's going on with the latest MariaDB Docker release.

confirmed. marking this as an answer

**Reply by JC5** [reply]
Thanks for the feedback, I'll check it out for sure. Perhaps I'll have to pin the previous release in the Docker file to be safe.

**Comment by t-hamano** [comment]
It may be related to the recent update of the MariaDB image. See https://github.com/MariaDB/mariadb-docker/issues/560

**Reply by JC5** [reply]
Indeed, the error message is the same.

**Comment by DarkWayC0de** [comment]
I had the same error, during the night I updated mariadb and fireflyiii stopped connecting to the mariadb database, I thought that with a mysql client I can connect to the database without any problem, but after reverting to the previous version firefly is working correctly again. It would be interesting to know what was the problem with the connection between fireflyiii and the new version of mariadb.

**Comment by ahmaddxb** [comment]
when firefly_iii_core is in an unhealthy state like this cause here, I can not stop or remove the container, I have to do a complete reboot and catch it early to remove / stop.

**Comment by dmarguerat** [comment]
Simply set Charest in docker compose will work.

command: --character-set-server=utf8

**Reply by xavasp** [reply]
Thanks, now it works.

**Reply by czwama** [reply]
Thanks, this fixed it for me as well

**Reply by Leah96xxx** [reply]
@dmarguerat which service do we need to add this command to? Firefly core or firefly db?

**Reply by dmarguerat** [reply]
@Leah96xxx Firefly db which is the MariaDB container.

**Reply by Leah96xxx** [reply]
Thanks. Confirmed that's fixed it for me too using mariadb:latest

**Comment by JC5** [comment]
In the meantime, the mariadb containers have been updated as well, so the problem should be solved either way. Thanks everybody for your patience and feedback!

**Reply by nottrobin** [reply]
But I'm doing this from scratch today, and I get the error

**Comment by DanielvdSpoel** [comment]
I'm having the exact same issue, but instead of MariaDB, I'm using MySQL, running on unraid using docker
